custom/enikshay/integrations/nikshay/repeaters.py

```diff
@@ -117,24 +117,22 @@ def get_custom_url(cls, domain):
 
     def allowed_to_forward(self, episode_case):
         allowed_case_types_and_users = self._allowed_case_type(episode_case) and self._allowed_user(episode_case)
+        if not allowed_case_types_and_users:
+            return False
+
         episode_case_properties = episode_case.dynamic_case_properties()
-        return allowed_case_types_and_users and (
+        if (
             episode_case_properties.get('nikshay_registered', 'false') == 'true' and
             episode_case_properties.get('nikshay_id', False) and
             case_properties_changed(episode_case, [TREATMENT_OUTCOME]) and
             episode_case_properties.get(TREATMENT_OUTCOME) in treatment_outcome.keys()
-        )
-
-
-def test_submission(person_case):
-    try:
-        phi_location = SQLLocation.objects.get(location_id=person_case.owner_id)
-    except SQLLocation.DoesNotExist:
-        raise NikshayLocationNotFound(
-            "Location with id {location_id} not found. This is the owner for person with id: {person_id}"
-            .format(location_id=person_case.owner_id, person_id=person_case.case_id)
-        )
-    return phi_location.metadata.get('is_test', "yes") == "yes"
+        ):
+            try:
+                person_case = get_person_case_from_episode(episode_case.domain, episode_case)
+            except ENikshayCaseNotFound:
+                return False
+            return not is_submission_from_test_location(person_case)
+        return False
```

Comment:
@proteusvacuum i think this is slightly more complicated than needed. We can just move this try block up and then have all conditions at once.
Though the change is definitely better since it avoids fetching an extra case and may be avoids a DB fetch in most of the situation. Just wondering if it written in a better manner. If not, then this its fine.